### ISSUE TITLE
[@medusajs/medusa-js] Incorrect return type from `lineItems.delete`

### ISSUE THREAD

**Issue by RATIU5** [medusajs/medusa#9571]
### Describe the bug

Take a look at this code in the code snippets section.

The type that's provided in the parameter of the `.then` callback from the delete method is of type `Response<StoreCartsRes>` which lists a `cart` as one of it's properties. This is incorrect. The `console.log` statement shows a different data structure, with the property `parent` containing the original cart object.

### System information

Medusa version (including plugins):

- `@medusajs/admin-sdk`: rc-4
- `@medusajs/framework`: rc-4
- `@medusajs/medusa`: rc-4
- `@medusajs/medusa-cli`: rc-4
- `@medusajs/medusa-js`: rc-4 (used on the storefront)

Node.js version: 22.9.0
Database: Postgres 16
Operating system: macOS 15
Browser (if relevant): Arc (Chromium)

### Steps to reproduce the behavior

1. Install the `rc` version of the `@medusajs/medusa-js` on your storefront. 
2. Connect your `rc-4` Medusa backend to the storefront.
3. Setup a Medusa client on the storefront and copy the code below.
4. Execute the code below on a line item delete click. See console for different data structure.

### Expected behavior

The expected type for `data` to have on the `delete` function's `.then` function is something like `StoreLineItemDeleteResponse`, which has a `parent` property which is the cart item. To keep the cart property, some rewrite of the `/store/carts/${cartId}/line-items/${lineItemId}` would be required. 
In other words, the expected return type of this `delete` function should match the REST API's endpoint.

### Code snippets

```ts
export const deleteCartItem = async (lineItemId: string, cartId: string) => {
  const headers = getMedusaHeaders();
  return medusaClient.carts.lineItems
    .delete(cartId, lineItemId, headers)
    .then((data) => {
      console.log(data); // `parent` property exists on the data object
      return data.cart; // `cart` does not exist on the data object (but is on the type of data)
    })
    .catch((err) => {
      console.log(err);
      return null;
    });
};
```

**Comment by olivermrbl** [comment]
`@medusajs/medusa-js` no longer exists. You should use `@medusajs/js-sdk` for Medusa 2.0.

Did you find a guide with this package somewhere? Maybe we need to get something updated.

**Comment by RATIU5** [comment]
Ah I see. I'm not sure which `rc` version npm is grabbing for `@medusajs/medusa-js`, I'll proceed with `@medusajs/js-sdk`. I think I was just following along with the old documentation for v1 since there isn't much documentation for the new `@medusajs/js-sdk`. Will update you if I find any mention of the old package. Thanks!
